doctrine/events.rst

```diff
@@ -0,0 +1,345 @@
+.. index::
+    single: Doctrine; Lifecycle Callbacks; Doctrine Events
+
+Doctrine Events
+===============
+
+`Doctrine`_, the set of PHP libraries used by Symfony to work with databases,
+provides a lightweight event system to update entities during the application
+execution. These events, called `lifecycle events`_, allow to perform tasks such
+as _"update the createdAt property automatically just before persisting entities
+of this type"_.
+
+Doctrine triggers events before/after performing the most common entity
+operations (e.g. ``prePersist/postPersist``, ``preUpdate/postUpdate``) and also
+on other common tasks (e.g. ``loadClassMetadata``, ``onClear``).
+
+There are two ways to listen to these Doctrine events:
+
+* **Lifecycle callbacks**, they are methods defined on the entity classes which
+  are called when the event is triggered;
+* **Lifecycle listeners and subscribers**, they are classes with specific callback
+  methods for one or more events.
+
+Callbacks have better performance because they are only called for the entities
+of a certain type, but can't reuse logic between different entities. Listeners
+and subscribers can reuse logic because they are called for all entities, but
+you need to add custom code to restrict the execution to certain entity types.
```

Comment:
There is a third way, available in newer versions of Doctrine ORM: entity listeners (rather than lifecycle listeners), which are registered for entities of a given type (with the filtering done by doctrine). Of couse, entity listeners can only listen to entity operations, not to other ORM operations.

Reply:
Thanks! I've added the "entity listener" docs and I'm proposing to remove them from the DoctrineBundle docs in https://github.com/doctrine/DoctrineBundle/pull/1019